### PR TITLE
Allow SSL connects to pgbouncer

### PR DESCRIPTION
**Pull request by dblackdblack**
### What does this PR do?
Rather than reconstructing the postgres connection string from its constituent parts (username, password, host, port), pass through connection strings authored by the user to psycopg2. This lets us use all of psycopg2's features, including SSL encryption.

### Motivation

pgbouncer v1.7 added SSL support, which is currently missing from the pgbouncer check.

### CODE REVIEW

**Comment by remh** [comment]
Thanks for doing this ! We'll review it soon.

File: checks.d/pgbouncer.py [hunk 1]
```diff
@@ -162,6 +168,7 @@ def check(self, instance):
         user = instance.get('username', '')
         password = instance.get('password', '')
         tags = instance.get('tags', [])
+        database_url = instance.get('database_url')
```
Comment (remh) [inline checks.d/pgbouncer.py]: Can you document that in the pgbouncer.yaml.example file please ?

Comment (dblackdblack) [inline checks.d/pgbouncer.py]: added

**Comment by dblackdblack** [comment]
@remh I rejiggered the connection code to be cleaner and to make it more obvious as when things are getting cached

File: checks.d/pgbouncer.py [hunk 2]
```diff
@@ -120,38 +120,62 @@ def _collect_stats(self, db, instance_tags):
...
+                host=host, port=port, user=user,
+                password=password, database_url=database_url
+            )
+
+            connection = pg.connect(**connect_kwargs)
+            connection.set_isolation_level(
+                pg.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+            self.log.debug('pgbouncer status: %s' % AgentCheck.OK)
+
+        # re-raise the ChceckExceptions raised by _get_connect_kwargs()
```
Comment (remh) [inline checks.d/pgbouncer.py]: Small typo here

File: checks.d/pgbouncer.py [hunk 3]
```diff
@@ -120,38 +120,62 @@ def _collect_stats(self, db, instance_tags):
...
+        except CheckException:
+            raise
+
+        except Exception:
+            message = u'Cannot establish connection to pgbouncer://%s:%s/%s' % (host, port, self.DB_NAME)
```
Comment (remh) [inline checks.d/pgbouncer.py]: I think if someoneuses the `database_url` to connect this message will be come
`Cannot establish connection to pgbouncer://:/` as it doesn't use the url at all.
Unless i'm missing something ?

File: checks.d/pgbouncer.py (hunk 1 above)
Comment (remh) [inline checks.d/pgbouncer.py]: Thanks 👍

**Comment by dblackdblack** [comment]
@remh pong!

**Comment by remh** [comment]
Thanks there is one last issue regarding the service check being sent when everything is alright https://github.com/plangrid/dd-agent/blob/0f5276b17725329803513cd7fb9e02ee4f44f0ed/checks.d/pgbouncer.py#L222 but i'll fix it.
Thanks again!

**Pull request merged**